src/main/java/org/mondemand/Client.java

```diff
@@ -454,14 +469,23 @@ public void addTransportsFromConfigFile(String configFileName)
    * Adds a new transport to this client.
    * @param transport the transport object to add
    */
-  public synchronized void addTransport(Transport transport) {
-    if (this.transports == null) {
-      this.transports = new Vector<Transport>();
+  public void addTransport(Transport transport) {
+    for (EventType eventType : EventType.values()) {
+      addTransport(eventType, transport);
     }
+  }
 
-    if (transport != null) {
-      this.transports.add(transport);
+  /**
+   * Adds a new event-specific transport to this client.
+   * @param eventType the event type for this transport
+   * @param transport the transport object to add
+   */
+  public void addTransport(EventType eventType, Transport transport) {
+    if (null == transport) {
+      return;
     }
+
+    this.transports.get(eventType).add(transport);
```

Comment:
i initialize an empty list for each event type in the constructor, so theoretically no.